### PR TITLE
add module backward

### PR DESCRIPTION
**Pull request by BBuf**
为以下Module添加Backward测试：

reshape
expand
squeeze
repeat

### CODE REVIEW

File: oneflow/core/framework/op_expr_helper.h [hunk 1]
```diff
@@ -180,5 +180,10 @@ Maybe<one::UserOpExpr> SparseSoftmaxCrossEntropyMsGradOp(const int64_t& depth,
 Maybe<one::UserOpExpr> PReLUGradOp();
 Maybe<one::UserOpExpr> PReLUGradOp(const std::string& name);
 
+Maybe<one::UserOpExpr> ExpandGradOp(const std::vector<int32_t> out_shape,
```
Comment (Flowingsun007) [inline oneflow/core/framework/op_expr_helper.h:183]: 是不是传引用&更好？

File: oneflow/python/test/modules/test_reshape.py [hunk 2]
```diff
@@ -15,8 +15,48 @@
 """
 import unittest
 
+from collections import OrderedDict
```
Comment (Flowingsun007) [inline oneflow/python/test/modules/test_reshape.py:18]: 根据import 规范改一下吧(import顺序为：标准库、第三方包、本地包，中间加空格)

File: oneflow/core/framework/op_expr_helper.h (hunk 1 above)
Comment (BBuf) [reply oneflow/core/framework/op_expr_helper.h:183]: 好的。

File: oneflow/python/test/modules/test_reshape.py (hunk 2 above)
Comment (BBuf) [reply oneflow/python/test/modules/test_reshape.py:18]: 好的。

File: oneflow/python/test/modules/test_repeat.py [hunk 3]
```diff
@@ -13,41 +13,126 @@
...
+    of_out = input.repeat(sizes=sizes)
+    np_out = np_repeat(input.numpy(), sizes)
+    test_case.assertTrue(np.array_equal(of_out.numpy(), np_out))
+
+
+def _test_repeat_same_dim_int(test_case, device):
+    input = flow.Tensor(
+        np.random.randn(1, 2, 5, 3), dtype=flow.int32, device=flow.device(device)
+    )
+    sizes = (4, 2, 3, 19)
```
Comment (Flowingsun007) [inline oneflow/python/test/modules/test_repeat.py:53]: 可以加一个用flow.Size构造的sizes，当作参数

File: oneflow/python/test/modules/test_reshape.py [hunk 4]
```diff
@@ -13,10 +13,51 @@
...
+    np_shape = (2, 2, 2, 2)
+    test_case.assertTrue(np.array_equal(of_shape, np_shape))
+
+
+def _test_tensor_reshape(test_case, device):
+    x = np.array(
+        [[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12], [13, 14, 15, 16]]
+    ).astype(np.float32)
+    input = flow.Tensor(x, device=flow.device(device))
+    of_shape = input.reshape(shape=[2, 2, 2, -1]).numpy().shape
```
Comment (Flowingsun007) [inline oneflow/python/test/modules/test_reshape.py:40]: shape参数需要支持tuple类型的吧？（需要加一个test case

File: oneflow/python/test/modules/test_squeeze.py [hunk 5]
```diff
@@ -13,10 +13,52 @@
...
+from test_util import GenArgList
+
+
+def _test_squeeze(test_case, device):
+    input = flow.Tensor(
+        np.array([[[[1, 1, 1]]]]).astype(np.int32), device=flow.device(device)
+    )
+    of_shape = flow.squeeze(input, dim=[1, 2]).numpy().shape
+    np_shape = (1, 3)
+    test_case.assertTrue(np.array_equal(of_shape, np_shape))
```
Comment (Flowingsun007) [inline oneflow/python/test/modules/test_squeeze.py:31]: testcase应该不仅仅是比较shape吧？（可以用np.sequeeze来对比squeeze后的值

File: oneflow/python/test/modules/test_reshape.py (hunk 4 above)
Comment (BBuf) [reply oneflow/python/test/modules/test_reshape.py:48]: 已解决。

File: oneflow/python/test/modules/test_squeeze.py (hunk 5 above)
Comment (BBuf) [reply oneflow/python/test/modules/test_squeeze.py:31]: 已解决。

**Review by Flowingsun007 — APPROVED** [review]

File: oneflow/python/nn/modules/expand.py [hunk 6]
```diff
@@ -65,7 +65,7 @@ def forward(self, x):
 @oneflow_export("expand")
 @register_tensor_op("expand")
 @experimental_api
-def expand_op(x, expand_size):
+def expand_op(x, *sizes):
```
Comment (doombeaker) [inline oneflow/python/nn/modules/expand.py:68]: 我看 PyTorch 的 `expand`， 只有 `torch.Tensor.expand`，没有 `torch.expand`。
我们是不是还做不到只提供 `torch.Tensor.expand`？

File: oneflow/core/framework/op_expr_helper.h [hunk 7]
```diff
@@ -195,4 +200,4 @@ Maybe<one::UserOpExpr> BinaryYGradOp(const std::string& binary_op_type);
 Maybe<one::UserOpExpr> BinaryYGradOp(const std::string& binary_op_type, const std::string& name);
 
 }  // namespace op_expr_helper
-}  // namespace oneflow
+}  // namespace oneflow
```
Comment (doombeaker) [inline oneflow/core/framework/op_expr_helper.h:203]: 文件末没有换行

File: oneflow/core/framework/op_expr_helper.cpp [hunk 8]
```diff
@@ -603,4 +617,4 @@ Maybe<one::UserOpExpr> BinaryYGradOp(const std::string& binary_op_type, const st
 }
 
 }  // namespace op_expr_helper
-}  // namespace oneflow
+}  // namespace oneflow
```
Comment (doombeaker) [inline oneflow/core/framework/op_expr_helper.cpp:620]: 文末换行

File: oneflow/python/test/modules/test_expand.py [hunk 9]
```diff
@@ -13,10 +13,13 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
-import unittest
+from collections import OrderedDict
 
+import unittest
```
Comment (doombeaker) [inline oneflow/python/test/modules/test_expand.py:18]: 放到第一类

File: oneflow/python/test/modules/test_expand.py [hunk 10]
```diff
@@ -77,46 +80,118 @@ def getOffset(i_offset, stride, expand_stride, n):
     return input, gout, out.reshape(tuple(new_size)), gin.reshape(input_shape)
 
 
+def _test_expand_new_dims(test_case, device):
+    input_shape = (1, 4, 1, 32)
+    expand_dim = [2, 1, 2, 4, 2, 32]
+    input, gout, out_np, gin_np = getExpandGrad(input_shape, expand_dim)
```
Comment (doombeaker) [inline oneflow/python/test/modules/test_expand.py:86]: 我觉得既然这里计算了梯度等。

可以在一个函数内，正向、反向的测试都做了。

否则此处调用 getExpandGrad，多出了多个不会使用的返回值，比较奇怪。

File: oneflow/python/test/modules/test_expand.py [hunk 11]
```diff
@@ -77,46 +80,118 @@ def getOffset(i_offset, stride, expand_stride, n):
...
+        input, dtype=flow.float32, device=flow.device(device), requires_grad=True
+    )
+    of_out = of_input.expand(2, 1, 2, 4, 2, 32)
+    test_case.assertTrue(np.array_equal(of_out.numpy(), out_np))
+
+
+def _test_expand_same_dim(test_case, device):
+    input_shape = (2, 4, 1, 32)
+    expand_dim = [2, 4, 2, 32]
+    input, gout, out_np, gin_np = getExpandGrad(input_shape, expand_dim)
```
Comment (doombeaker) [inline oneflow/python/test/modules/test_expand.py:97]: 同上，可以考虑前向、后向的测试合并一起做。

File: oneflow/python/test/modules/test_expand.py (hunk 10 above)
Comment (doombeaker) [inline oneflow/python/test/modules/test_expand.py:86]: `getExpandGrad` 不是很符合 python 的常见命名习惯（我没看 PEP8， 但是应该是不符合的），可以考虑改成 `_get_expand_grads`

File: oneflow/python/test/modules/test_expand.py [hunk 12]
```diff
@@ -77,46 +80,118 @@ def getOffset(i_offset, stride, expand_stride, n):
...
+    of_input = flow.Tensor(input, dtype=flow.float32, device=flow.device(device))
+    of_out = of_input.expand(2, 4, 2, 32)
+
+    test_case.assertTrue(np.array_equal(of_out.numpy(), out_np))
+
+
+def _test_expand_same_dim_negative(test_case, device):
```
Comment (doombeaker) [inline oneflow/python/test/modules/test_expand.py:104]: 可以考虑把 `shape`, `dim` 提取出来，同device一样作为参数，这样 `_test_expand_same_dim`, `_test_expand_same_dim_negative`, `_test_expand_same_int` 可以合并成一个。

可以参考： https://github.com/Oneflow-Inc/oneflow/blob/master/oneflow/python/test/modules/test_exp.py

File: oneflow/python/test/modules/test_expand.py [hunk 13]
```diff
@@ -77,46 +80,118 @@ def getOffset(i_offset, stride, expand_stride, n):
...
+    of_input = flow.Tensor(input, dtype=flow.int, device=flow.device(device))
+    of_out = of_input.expand(2, 4, 2, 32)
+
+    test_case.assertTrue(np.array_equal(of_out.numpy(), out_np.astype(np.int32)))
+
+
+def _test_expand_backward_same_dim(test_case, device):
+    input_shape = (2, 4, 1, 1)
+    expand_dim = [2, 4, 2, 1]
+    input = np.array(
```
Comment (doombeaker) [inline oneflow/python/test/modules/test_expand.py:127]: 我看 `getExpandGrad` 里是有 np 模拟过程的，这种数据写死的写法可以去掉了？

File: oneflow/python/nn/modules/expand.py (hunk 6 above)
Comment (BBuf) [reply oneflow/python/nn/modules/expand.py:68]: 可以做到，已修改。

File: oneflow/core/framework/op_expr_helper.cpp (hunk 8 above)
Comment (BBuf) [reply oneflow/core/framework/op_expr_helper.cpp:620]: 这些格式问题之前都调整过，是因为手动合并冲突产生的，已修改。

File: oneflow/python/test/modules/test_expand.py (hunk 10 above)
Comment (BBuf) [reply oneflow/python/test/modules/test_expand.py:86]: 这里算出来的梯度有错误，那我把这个函数删了吧。

File: oneflow/python/test/modules/test_expand.py (hunk 9 above)
Comment (BBuf) [reply oneflow/python/test/modules/test_expand.py:18]: 好的

File: oneflow/python/test/modules/test_expand.py (hunk 13 above)
Comment (BBuf) [reply oneflow/python/test/modules/test_expand.py:127]: 同一个问题，这个梯度模拟是错误的，删除这个函数。

File: oneflow/python/test/modules/test_expand.py (hunk 12 above)
Comment (BBuf) [reply oneflow/python/test/modules/test_expand.py:104]: 由于梯度模拟目前numpy实现存在和pytorch无法对齐的原因，先保留这种固定输入的做法。

File: oneflow/python/test/modules/test_squeeze.py [hunk 14]
```diff
@@ -13,10 +13,73 @@
...
+import unittest
 import numpy as np
+
 import oneflow.experimental as flow
+from test_util import GenArgList
+
+
+def _test_squeeze(test_case, device):
+    input = flow.Tensor(
+        np.array([[[[1, 1, 1]]]]).astype(np.int32), device=flow.device(device)
```
Comment (Flowingsun007) [inline oneflow/python/test/modules/test_squeeze.py:27]: input建议都改成np随机矩阵的方式（np不好造、可以用硬编码的方式
